pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -89,9 +87,10 @@ protected String getQuery(SlingHttpServletRequest request) {
         }
 
         StringBuilder queryBuilder = new StringBuilder()
-                .append("/jcr:root/content/(repositories | modules)//element(*, pant:module)");
+                .append("/jcr:root/content/(repositories | modules)//element()");
 
-        List<StringBuilder> queryFilters = newArrayListWithCapacity(4);
+        List<StringBuilder> queryFilters = newArrayListWithCapacity(5);
+        queryFilters.add(new StringBuilder("(@jcr:primaryType = 'pant:assembly' or @jcr:primaryType = 'pant:module')"));
```

Comment:
I see. We **could** create a similar structure to our java code where we have [pant:document] and have [pant:assembly] and [pant:module] extend from it. Then we could do the simpler query.

But I don't know if it's worth it since we are keeping those JCR types very simple.

Reply:
(not a blocker)

Reply:
Yeah, that same thought crossed my mind as well. I didn't like the idea of creating pant:document as an explicit node type, because then we are in danger of accidentally creating pant:document nodes in the jcr, and in that scenario, we would definitely be doing something wrong.

This query could potentially be improved, but in the spirit of "don't let perfect be the enemy of good," I say we leave this as-is for now and revisit it in the future if it becomes problematic for some reason.